wolfcrypt/src/asn.c

```diff
@@ -31698,9 +31698,12 @@ int StoreECC_DSA_Sig(byte* out, word32* outLen, mp_int* r, mp_int* s)
     }
     if (ret == 0) {
         /* Encode DSA signature into buffer. */
-        SetASN_Items(dsaSigASN, dataASN, dsaSigASN_Length, out);
-        /* Set the actual encoding size. */
-        *outLen = (word32)sz;
+        ret = SetASN_Items(dsaSigASN, dataASN, dsaSigASN_Length, out);
+        if ((ret >= 0) && (ret == sz)) {
```

Comment:
It looks like if SetASN_Items return value for some reason did not match the 'sz' value from SizeASN_Items then 'outLen' would not be set and a positive value would be returned (neither success or any expected error value). Can we add an error case for ret >=0 and ret != sz?